subprojects/ide/src/main/java/org/gradle/plugins/ide/eclipse/EclipsePlugin.java

```diff
@@ -368,14 +409,17 @@ public void execute(Plugin<Project> plugin) {
         };
     }
 
-    private static <T extends Task> void maybeAddTask(Project project, IdePlugin plugin, String taskName, Class<T> taskType, Action<T> action) {
+    private static <T extends Task> TaskProvider<T> maybeAddTask(Project project, IdePlugin plugin, String taskName, Class<T> taskType, Action<T> action) {
         TaskContainer tasks = project.getTasks();
-        if (tasks.findByName(taskName) != null) {
-            return;
+        T taskFound = tasks.withType(taskType).findByName(taskName);
```

Comment:
This will (or should) realise an already registered task. So, we should either 1. add a way to find by name without realising the task or 2. just get rid of this lookup. Why should the plugin assume that a task that happens to exist with the same names that it uses happens to the right thing? Seems like a bad assumption to make.